### ISSUE TITLE
MathJax deletion can not be undone

### ISSUE THREAD

**Issue by dae** [ankitects/anki#1941]
If the entire equation is accidentally deleted with backspace, ctrl+z does not recover it.

https://forums.ankiweb.net/t/backspacing-an-equation-cannot-be-undone-with-ctrl-z/21103

**Comment by pwintz** [comment]
(In Version ⁨2.1.53 on Windows), this issue occurs if you delete an _inline_ MathJax equation using backspace or the delete button in the math editor popup (see screenshot). If you delete a block equation using backspace, then undo works, but not if you use the delete button.

![image](https://user-images.githubusercontent.com/19618780/176842545-a67e689f-c17c-41e6-9cc4-5450580582ad.png)

**Comment by dae** [comment]
Issues with undoing cloze-deleted MathJax too: 
https://forums.ankiweb.net/t/mangled-cloze-deletion-when-cloze-includes-mathjax/21121

**Comment by dae** [comment]
Another perhaps related issue: https://forums.ankiweb.net/t/ctrl-z-bug-about-mathjax-within-the-same-p-tag/26236

**Comment by s9gf4ult** [comment]
Still relevant. Very annoying. This especially clashes with cloze deletion cards, because sometimes `Ctrl-Shift-C` removes formulas and it can not be undone.

**Comment by romankrru** [comment]
I agree that the problem is really annoying. What I do as a workaround is switch to the HTML editor and press CTRL-Z directly in it.
![image](https://github.com/user-attachments/assets/f4025842-8279-43c2-897d-c656ddebff04)

**Comment by khonkhortisan** [comment]
[Anki Version ⁨25.02 (038d85b1)⁩] I can even backspace mathjax AFTER the caret

![Image](https://github.com/user-attachments/assets/0eba9c0f-3a12-4904-a81b-f4b941ac6bcb)

This field started as `<p><anki-mathjax>5\sqrt{2}+2\sqrt{10}</anki-mathjax></p>` and I added the mathjax, then pressed backspace to reduce the spacing, which deletes the mathjax after it in such a way that ctrl-z undo doesn't work. If it had started as `<br>` instead of `<p></p>`, it would've worked. This happens with and without addons.
```html
<p><anki-mathjax>5\sqrt{2}+2\sqrt{10}</anki-mathjax></p><p></p><p><anki-mathjax>\frac{\sqrt{10}}{\sqrt5-2}=\frac{\sqrt{10}}{\sqrt5-2}\cdot\frac{\sqrt5-2}{\sqrt5-2}=\frac{\sqrt{50}+2\sqrt{10}}{(\sqrt5)^2-2^2}=\frac{5\sqrt2+2\sqrt{10}}{5-4}=5\sqrt2+2\sqrt10</anki-mathjax><br></p>
```
